<center>
<img src="../../img/ods_stickers.jpg" />
    
## [mlcourse.ai](mlcourse.ai) – Open Machine Learning Course 
Author: Arina Lopukhova (@erynn). Edited by [Yury Kashnitskiy](https://yorko.github.io) (@yorko). This material is subject to the terms and conditions of the [Creative Commons CC BY-NC-SA 4.0](https://creativecommons.org/licenses/by-nc-sa/4.0/) license. Free use is permitted for any non-commercial purpose.

<h1><center>Assignment #1. Fall 2018</center></h1>
<h2><center>Exploratory data analysis of Olympic games with Pandas</center></h2>

__There are ten questions about [120 years of Olympic history: athletes and results](https://www.kaggle.com/heesoo37/120-years-of-olympic-history-athletes-and-results/version/2) dataset in this task. Your task is to fill in the missing Python code and choose answers in [this web-form](https://docs.google.com/forms/d/179HYvlF_0x435CWdu2aOtR9a3flh0YCQXXNXhXaKw2M).__

Download the file `athlete_events.csv` from [Kaggle page](https://www.kaggle.com/heesoo37/120-years-of-olympic-history-athletes-and-results). The dataset has the following features (copied from Kaggle):

- __ID__ - Unique number for each athlete
- __Name__ - Athlete's name
- __Sex__ - M or F
- __Age__ - Integer
- __Height__ - In centimeters
- __Weight__ - In kilograms
- __Team__ - Team name
- __NOC__ - National Olympic Committee 3-letter code
- __Games__ - Year and season
- __Year__ - Integer
- __Season__ - Summer or Winter
- __City__ - Host city
- __Sport__ - Sport
- __Event__ - Event
- __Medal__ - Gold, Silver, Bronze, or NA

In [89]:
import pandas as pd
import numpy as np

In [4]:
# Change the path to the dataset file if needed. 
PATH = 'athlete_events.csv'

In [5]:
data = pd.read_csv(PATH)
data.head()

ID                      Name Sex   Age  Height  Weight            Team  \
0   1                 A Dijiang   M  24.0   180.0    80.0           China   
1   2                  A Lamusi   M  23.0   170.0    60.0           China   
2   3       Gunnar Nielsen Aaby   M  24.0     NaN     NaN         Denmark   
3   4      Edgar Lindenau Aabye   M  34.0     NaN     NaN  Denmark/Sweden   
4   5  Christine Jacoba Aaftink   F  21.0   185.0    82.0     Netherlands   

   NOC        Games  Year  Season       City          Sport  \
0  CHN  1992 Summer  1992  Summer  Barcelona     Basketball   
1  CHN  2012 Summer  2012  Summer     London           Judo   
2  DEN  1920 Summer  1920  Summer  Antwerpen       Football   
3  DEN  1900 Summer  1900  Summer      Paris     Tug-Of-War   
4  NED  1988 Winter  1988  Winter    Calgary  Speed Skating   

                              Event Medal  
0       Basketball Men's Basketball   NaN  
1      Judo Men's Extra-Lightweight   NaN  
2           Football Men's Football   NaN  
3       Tug-Of-War Men's Tug-Of-War  Gold  
4  Speed Skating Women's 500 metres   NaN

In [34]:
# browser the data frame 
print(data.describe(include = "all"))
data.info()

                   ID                  Name     Sex            Age  \
count   271116.000000                271116  271116  261642.000000   
unique            NaN                134732       2            NaN   
top               NaN  Robert Tait McKenzie       M            NaN   
freq              NaN                    58  196594            NaN   
mean     68248.954396                   NaN     NaN      25.556898   
std      39022.286345                   NaN     NaN       6.393561   
min          1.000000                   NaN     NaN      10.000000   
25%      34643.000000                   NaN     NaN      21.000000   
50%      68205.000000                   NaN     NaN      24.000000   
75%     102097.250000                   NaN     NaN      28.000000   
max     135571.000000                   NaN     NaN      97.000000   

               Height         Weight           Team     NOC        Games  \
count   210945.000000  208241.000000         271116  271116       271116   
unique 

In [38]:
# combine multy condiction, using parathenses to combine different conditions. 
print(data[(data.Year == 1996) & (data.Sex == "M")]["Age"].min())  
print(data[(data.Year == 1996) & (data.Sex == "F")]["Age"].min())

14.0
12.0


__1. How old were the youngest male and female participants of the 1996 Olympics?__

- 16 and 15
- 14 and 12 
- 16 and 12
- 13 and 11

In [44]:
data["Sport"].value_counts()

Athletics                38624
Gymnastics               26707
Swimming                 23195
Shooting                 11448
Cycling                  10859
Fencing                  10735
Rowing                   10595
Cross Country Skiing      9133
Alpine Skiing             8829
Wrestling                 7154
Football                  6745
Sailing                   6586
Equestrianism             6344
Canoeing                  6171
Boxing                    6047
Speed Skating             5613
Ice Hockey                5516
Hockey                    5417
Biathlon                  4893
Basketball                4536
Weightlifting             3937
Water Polo                3846
Judo                      3801
Handball                  3665
Art Competitions          3578
Volleyball                3404
Bobsleigh                 3058
Tennis                    2862
Diving                    2842
Ski Jumping               2401
                         ...  
Badminton                 1457
Nordic C

In [50]:
# You code here
a1 = data[(data.Year == 2000) & (data.Sport == "Gymnastics") & (data.Sex == "M")].shape
a2 = data[(data.Year == 2000) & (data.Sex == "M")].shape
a1[0] / a2[0]

0.07651966626936829

__2. What was the percentage of male gymnasts among all the male participants of the 2000 Olympics? Round the answer to the first decimal.__

*Hint:* here and further if needed drop duplicated sportsmen to count only unique ones. 

- 0.2
- 1.5 
- 2.5
- 7.7

In [53]:
# You code here 
data[(data.Sport=="Basketball") & (data.Sex=="F") & (data.Year == 2000)].Height.describe()

count    142.000000
mean     182.387324
std        9.139462
min      162.000000
25%      175.000000
50%      182.000000
75%      190.000000
max      213.000000
Name: Height, dtype: float64

__3. What are the mean and standard deviation of height for female basketball players participated in the 2000 Olympics? Round the answer to the first decimal.__

- 178.5 and 7.2
- 179.4 and 10
- 180.7 and 6.7
- 182.4 and 9.1 

In [58]:
# You code here 
highestW = data[(data.Year==2002)]["Weight"].max()
data[(data.Year==2002)&(data.Weight== highestW)]["Sport"]

99154    Bobsleigh
Name: Sport, dtype: object

__4. Find a sportsperson participated in the 2002 Olympics, with the highest weight among other participants of the same Olympics. What sport did he or she do?__

- Judo
- Bobsleigh 
- Weightlifting
- Boxing

In [66]:
# You code here 
data[data.Name=="Pawe Abratkiewicz"]["Year"].value_counts()

1998    2
2002    2
1992    2
Name: Year, dtype: int64

__5. How many times did Pawe Abratkiewicz participate in the Olympics held in different years?__

- 0
- 1 
- 2
- 3 

In [69]:
# You code here 
data[(data.Year==2000)&(data.Sport=="Tennis")&(data.Team=="Australia") &(data.Medal =="Silver")]

ID                    Name Sex   Age  Height  Weight       Team  \
262820  131504  Todd Andrew Woodbridge   M  29.0   178.0    75.0  Australia   
262831  131511  Mark Raymond Woodforde   M  34.0   183.0    80.0  Australia   

        NOC        Games  Year  Season    City   Sport                 Event  \
262820  AUS  2000 Summer  2000  Summer  Sydney  Tennis  Tennis Men's Doubles   
262831  AUS  2000 Summer  2000  Summer  Sydney  Tennis  Tennis Men's Doubles   

         Medal  
262820  Silver  
262831  Silver

__6. How many silver medals in tennis did Australia win at the 2000 Olympics?__

- 0
- 1 
- 2 
- 3 

In [86]:
# You code here 
a1 = data[(data.Team=="Switzerland") & (data.Year==2016)]["Medal"].dropna().count()
a2 = data[(data.Team=="Serbia") & (data.Year==2016)]["Medal"].dropna().count()
a1 < a2

True

__7. Is it true that Switzerland won fewer medals than Serbia at the 2016 Olympics? Do not consider NaN values in _Medal_ column.__ 

- Yes 
- No

In [105]:
# You code here 
select = data[(data.Year == 2014)]
ran = np.arange(5, 100, 10)
# df.groupby(pd.cut(df.a, ranges)).count()
select.groupby(pd.cut(select.Age, ran)).count()

ID  Name   Sex   Age  Height  Weight  Team   NOC  Games  Year  \
Age                                                                         
(5, 15]     10    10    10    10      10       8    10    10     10    10   
(15, 25]  2412  2412  2412  2412    2396    2254  2412  2412   2412  2412   
(25, 35]  2281  2281  2281  2281    2277    2227  2281  2281   2281  2281   
(35, 45]   184   184   184   184     184     181   184   184    184   184   
(45, 55]     4     4     4     4       4       3     4     4      4     4   
(55, 65]     0     0     0     0       0       0     0     0      0     0   
(65, 75]     0     0     0     0       0       0     0     0      0     0   
(75, 85]     0     0     0     0       0       0     0     0      0     0   
(85, 95]     0     0     0     0       0       0     0     0      0     0   

          Season  City  Sport  Event  Medal  
Age                                          
(5, 15]       10    10     10     10      3  
(15, 25]    2412  2412   2412   2412    250  
(25, 35]    2281  2281   2281   2281    316  
(35, 45]     184   184    184    184     28  
(45, 55]       4     4      4      4      0  
(55, 65]       0     0      0      0      0  
(65, 75]       0     0      0      0      0  
(75, 85]       0     0      0      0      0  
(85, 95]       0     0      0      0      0

__8. What age category did the fewest and the most participants of the 2014 Olympics belong to?__
- [45-55] and [25-35) correspondingly
- [45-55] and [15-25) correspondingly
- [35-45] and [25-35) correspondingly
- [45-55] and [35-45) correspondingly

In [110]:
# You code here 
data[(data.Season == "Summer") & (data.City =="Lake Placid")].count()
data[(data.Season == "Winter") & (data.City =="Sankt Moritz")].count()

ID        1657
Name      1657
Sex       1657
Age       1563
Height     238
Weight     159
Team      1657
NOC       1657
Games     1657
Year      1657
Season    1657
City      1657
Sport     1657
Event     1657
Medal      224
dtype: int64

__9. Is it true that there were Summer Olympics held in Lake Placid? Is it true that there were Winter Olympics held in Sankt Moritz?__

- Yes, Yes
- Yes, No
- No, Yes 
- No, No 

In [120]:
# You code here 
# there is no olimpics in 1995, there IS Olympics in 1996
a = data[(data.Year== 2016)].Sport.value_counts()
a.shape

(34,)

__10. What is the absolute difference between the number of unique sports at the 1995 Olympics and 2016 Olympics?__

- 16
- 24
- 26
- 34